src/lib/mina_base/transaction_union_payload.ml

```diff
@@ -164,15 +138,20 @@ module Body = struct
       { tag = Tag.unpacked_of_t tag
       ; source_pk = Public_key.Compressed.var_of_t source_pk
       ; receiver_pk = Public_key.Compressed.var_of_t receiver_pk
-      ; token_id = Token_id.var_of_t token_id
+      ; token_id = Tid.Checked.constant token_id
       ; amount = Currency.Amount.var_of_t amount
       ; token_locked = Boolean.var_of_value token_locked
       }
 
     let to_input_legacy
         { tag; source_pk; receiver_pk; token_id; amount; token_locked } =
```

Comment:
Can we drop the `token_id` field from this record?

Reply:
We can't at the moment because it's used for representing fee transfers and for `Two` fee transfers, we pack them into the payload record by using the `token_id` field of `body` and the `fee_token` field of `common` to represent the two.

Reply:
We probably will need to rewrite a lot of this logic if/when we get around to adding non-default fee tokens, but I don't want to have to make those changes now.